src/phylum/init/cli.py

```diff
@@ -0,0 +1,270 @@
+
+    # Check that the token was setup correctly by using it to display the current auth status
+    cmd_line = [PHYLUM_BIN_PATH, "auth", "status"]
+    subprocess.run(cmd_line, check=True)
+
+
+def get_args():
+    """Get the arguments from the command line and return them."""
+    parser = argparse.ArgumentParser(
+        prog=SCRIPT_NAME,
+        description="Fetch and install the Phylum CLI",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "-v",
+        "--phylum-version",
+        dest="version",
+        default="latest",
+        type=version_check,
+        help="the version of the Phylum CLI to install",
+    )
+    parser.add_argument(
+        "-t",
+        "--target",
+        choices=SUPPORTED_TARGET_TRIPLES,
+        default=get_target_triple(),
+        help="the target platform type where the CLI will be installed",
+    )
+    parser.add_argument(
+        "-k",
+        "--phylum-token",
+        dest="token",
+        help=f"""Phylum user token. Can also specify this option's value by setting the `{TOKEN_ENVVAR_NAME}`
+            environment variable. The value specified with this option takes precedence when both are provided.
+            Leave this option and it's related environment variable unspecified to either (1) use an existing token
+            already set in the Phylum config file or (2) to manually populate the token with a `phylum auth login` or
+            `phylum auth register` command after install.""",
+    )
+    parser.add_argument(
+        "--version",
+        action="version",
+        version=f"{SCRIPT_NAME} {__version__}",
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    """Main entrypoint."""
+    args = get_args()
+
+    target_triple = args.target
+    if target_triple not in SUPPORTED_TARGET_TRIPLES:
+        raise ValueError(f"The identified target triple `{target_triple}` is not currently supported")
+
+    archive_name = f"phylum-{target_triple}.zip"
+    minisig_name = f"{archive_name}.minisig"
+    archive_url = get_archive_url(args.version, archive_name)
+    minisig_url = f"{archive_url}.minisig"
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_dir_path = pathlib.Path(temp_dir)
+        archive_path = temp_dir_path / archive_name
+        minisig_path = temp_dir_path / minisig_name
+
+        save_file_from_url(archive_url, archive_path)
+        save_file_from_url(minisig_url, minisig_path)
+
+        verify_minisig(archive_path, minisig_path)
+
+        with zipfile.ZipFile(archive_path, mode="r") as zip_file:
+            if zip_file.testzip() is not None:
+                raise zipfile.BadZipFile(f"There was a bad file in the zip archive {archive_name}")
+            extracted_dir = temp_dir_path
+            top_level_zip_entry = zip_file.infolist()[0]
+            if top_level_zip_entry.is_dir():
+                extracted_dir = temp_dir_path / top_level_zip_entry.filename
```

Comment:
This requires that the directory is the first entry in the zip file, which seems like an odd requirement, especially considering that it will just happily continue if it's not a directory. And considering this directory should always have the same name as the archive.

```suggestion
            extracted_dir = temp_dir_path / f"phylum-{target_triple}"
```

Reply:
The idea with the original code was to account for zip file structures where the contents get extracted into either a directory _or_ all together at the same level as the zip file itself. Granted, in this case, the structure is controlled by Phylum...with the CLI release process...and currently will be extracted into a directory with a known name.

FWIW, the [docs for `Zipfile.infolist()`](https://devdocs.io/python~3.10/library/zipfile#zipfile.ZipFile.infolist) say:

> Return a list containing a [ZipInfo](https://devdocs.io/python~3.10/library/zipfile#zipfile.ZipInfo) object for each member of the archive. The objects are in the same order as their entries in the actual ZIP file on disk if an existing archive was opened.

Reply:
Right. It's just a bit brittle. What if the first thing in the zip file happens to be the `completions` folder?

Reply:
ooh...now I see it! The change has been made.